src/PowerShell/AzPSLogin.ts

```diff
@@ -0,0 +1,93 @@
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+import * as os from 'os';
+import * as path from 'path';
+
+import AzPSScriptBuilder from './AzPSScriptBuilder';
+import AzPSConstants from './AzPSConstants';
+import { LoginConfig } from '../common/LoginConfig';
+
+export class AzPSLogin {
+    loginConfig: LoginConfig;
+
+    constructor(loginConfig: LoginConfig) {
+        this.loginConfig = loginConfig;
+    }
+
+    async login() {
+        core.info(`Running Azure PowerShell Login.`);
+        this.setPSModulePathForGitHubRunner();
+        await this.importLatestAzAccounts();
+
+        const [loginMethod, loginScript] = await AzPSScriptBuilder.getAzPSLoginScript(this.loginConfig);
+        core.info(`Attempting Azure PowerShell login by using ${loginMethod}...`);
+        core.debug(`Azure PowerShell Login Script: ${loginScript}`);
+        await AzPSLogin.runPSScript(loginScript);
+        console.log(`Running Azure PowerShell Login successfully.`);
+    }
+
+    setPSModulePathForGitHubRunner() {
+        const runner: string = process.env.RUNNER_OS || os.type();
+        switch (runner.toLowerCase()) {
+            case "linux":
+                this.pushPSModulePath(AzPSConstants.DEFAULT_AZ_PATH_ON_LINUX);
```

Comment:
Yes. It'll also work on a self-hosted runner. 
There are 2 options for this:

- Only apply this for GitHub-Hosted runners. 
  - I see there is an env property **RUNNER_ENVIRONMENT** which have 2 values "self-hosted" and "github-hosted". But I didn't find official docs for it. That's why I don't suggest to use it in our code since it's not reliable. I didn't find other ways to check if this action is running on a GitHub-Hosted runner.
- The other way is to keep it as it is now.
  - If so, this line will be applied for "self-hosted" runner too. It could add an invalid path to PSModulePath. However, it does no harm. We have test cases for it.

I suggest we choose option 2.